<!--
 Copyright (c) 2023  Carnegie Mellon University, IBM Corporation, and others

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
-->

<launch>
  <arg name="output" default="screen"/>
  <arg name="db_path" />
  <arg name="init_speed" default=""/>
  <arg name="config_user" value="cabot"/>
  <arg name="language" default="en"/>
  <arg name="action_name" default="/move_base" />
  <arg name="global_map_name" default="map" />
  <arg name="plan_topic" default="/move_base/NavfnROS/plan" />

  <arg name="use_skip_notify_similar_turn_directions" default="true"/>
  <arg name="distance_keep_similar_turn_directions" default="3.0"/>
  <arg name="max_abs_yaw_diff_similar_turn_directions" default="$(eval 3.1415 / 4.0)"/>
  <arg name="should_read_intersection_path_in_clock_direction" default="false"/>

  <arg name="use_skip_notify_backward_directions" default="true"/>
  <arg name="max_abs_yaw_diff_skip_backward_directions" default="$(eval 3.1415 / 4.0)"/>

  <arg name="use_skip_notify_short_route" default="true"/>
  <arg name="max_skip_short_route_distance" default="5.0"/>

  <include file="$(find mongodb_store)/launch/mongodb_store.launch">
    <arg name="db_path" value="$(arg db_path)"/>
  </include>

  <group ns="cabot">
    <node pkg="cabot_ui" type="cabot_ui_manager_explore.py"
    name="cabot_ui_manager_explore"
    output="$(arg output)">
      <rosparam file="$(find cabot_ui)/menu/menu.yaml" command="load"/>

      <param name="config_user" value="$(arg config_user)"/>
      <param name="init_speed" value="$(arg init_speed)"/>
      <param name="language" value="$(arg language)" />
      <param name="action_name" value="$(arg action_name)" />
      <param name="global_map_name" value="$(arg global_map_name)" />
      <param name="plan_topic" value="$(arg plan_topic)" />

      <param name="use_skip_notify_similar_turn_directions" value="$(arg use_skip_notify_similar_turn_directions)" />
      <param name="distance_keep_similar_turn_directions" value="$(arg distance_keep_similar_turn_directions)" />
      <param name="max_abs_yaw_diff_similar_turn_directions" value="$(arg max_abs_yaw_diff_similar_turn_directions)" />
      <param name="should_read_intersection_path_in_clock_direction" value="$(arg should_read_intersection_path_in_clock_direction)" />

      <param name="use_skip_notify_backward_directions" value="$(arg use_skip_notify_backward_directions)" />
      <param name="max_abs_yaw_diff_skip_backward_directions" value="$(arg max_abs_yaw_diff_skip_backward_directions)" />

      <param name="use_skip_notify_short_route" value="$(arg use_skip_notify_short_route)" />
      <param name="max_skip_short_route_distance" value="$(arg max_skip_short_route_distance)" />
    </node>
  </group>
</launch>
